corgi/tasks/prod_defs.py

```
@@ -67,6 +69,87 @@ def update_products() -> None:
                 parse_product_version(pd_product_version, product, product_node)
 
 
+def _find_by_cpe(cpe_patterns: list[str]) -> list[str]:
+    """Given a list of CPE patterns find all the CPEs from ET Variants which match it"""
+    if not cpe_patterns:
+        return []
+    regex_query = Q()
+    for pattern in cpe_patterns:
+        regex = pattern.replace(".", "\\.").replace("*", ".*")
+        regex_query |= Q(cpe__regex=regex)
+    return list(
+        CollectorErrataProductVariant.objects.filter(live=True)
+        .filter(regex_query)
+        .values_list("cpe", flat=True)
+    )
+
+
+def _match_stream_version_to_cpe_version(
+    version_streams: QuerySet[ProductStream],
+    cpe_version: str,
+    cpe: str,
+    el_by_cpe: defaultdict[str, list],
+    cpes_by_stream: defaultdict[str, list],
+):
+    """Iterate product stream versions, looking for matches to the cpe_version passed in.
+    If a match is found, store it in cpes_by_streams dict along with any el suffixes."""
+    for version_stream in version_streams:
+        if version_stream.version.removesuffix(".z") == cpe_version:
+            if cpe in el_by_cpe:
+                for el in el_by_cpe[cpe]:
+                    cpes_by_stream[version_stream.name].append(cpe + "::" + el)
+            else:
+                cpes_by_stream[version_stream.name].append(cpe)
+
+
+def _split_cpe_version(cpe_part: str, versions_by_cpe: defaultdict[str, set]):
+    version_split = cpe_part.rsplit(":", 1)
+    cpe_name = version_split[0]
+    cpe_version = version_split[-1]
```

Using `version_split[-1]` avoids an error when the length of version_split is only 1 / when a : colon wasn't found in the cpe_part. Is that on purpose, or do we need to raise an error in this case?

Currently both cpe_name and cpe_version will both be the same in this case, so the dict key would be "something_without_a_colon" and we'd also append "something_without_a_colon" to the list of values.

---

Not having a ':' int he cpe_part should never happen. But I updated the code here to check for that log a warning, and skip updating the versions_by_cpe map for that cpe.